aspnetcore/signalr/hub-filters.md

```diff
@@ -0,0 +1,203 @@
+---
+title: Use hub filters in ASP.NET Core SignalR
+author: brecon
+description: Learn how to use hub filters in ASP.NET Core SignalR.
+monikerRange: '>= aspnetcore-5.0'
+ms.author: brecon
+ms.custom: mvc
+ms.date: 05/22/2020
+no-loc: [Blazor, "Identity", "Let's Encrypt", Razor, SignalR]
+uid: signalr/hub-filters
+---
+
+# Use hub filters in ASP.NET Core SignalR
+
+Hub filters:
+
+* Are available in ASP.NET Core 5.0 or later.
+* Allow logic to run before and after hub methods are invoked by clients.
+
+This article provides guidance for writing and using hub filters.
+
+## Configure hub filters
+
+Hub filters can be applied globally or per hub type. The order in which filters are added is the order in which the filters run. Global hub filters run before local hub filters.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddSignalR(options =>
+    {
+        // Global filters will run first
+        options.AddFilter<CustomFilter>();
+    }).AddHubOptions<MyHub>(options =>
+    {
+        // Local filters will run second
+        options.AddFilter<CustomFilter2>();
+    });
+}
+```
+
+A hub filter can be added in one of the following ways:
+
+* Add a filter by concrete type:
+
+    ```csharp
+    hubOptions.AddFilter<TFilter>();
+    ```
+
+    This will be resolved from DI or type activated.
```

Comment:
Define DI the first time it's referenced in the doc.

```suggestion
    This will be resolved from dependency injection (DI) or type activated.
```